src/main/java/group2/cms/service/AuthorService.java

```diff
@@ -23,47 +27,66 @@
     @Autowired
     private AuthorRepository authorRepository;
 
-    public Author addAuthor(Long userID, Long paperID){
+    public AuthorDTO addAuthor(AuthorDTO authorDTO){
+        var authorData = converter.dtoToEntity(authorDTO);
+        var userID = authorData.getId();
+        var paperID = authorData.getPaper().getId();
         var paperUploader = userRepository.findById(userID).orElseThrow(
                 () -> new InvalidIDException("Invalid User ID: " + userID)
         );
         var uploadedPaper = paperRepository.findById(paperID).orElseThrow(
                 () -> new InvalidIDException("Invalid Paper ID: " + paperID)
         );
 
-        return authorRepository.save(new Author(paperUploader, uploadedPaper));
+        var addedAuthor =  authorRepository.save(new Author(paperUploader, uploadedPaper));
+        return converter.entityToDto(addedAuthor);
     }
 
-    public void deleteAuthor(Long authorID){
+    public void deleteAuthor(Long authorID)
+    {
         authorRepository.deleteById(authorID);
     }
 
-    public List<Author> getAllAuthors(){
-        return authorRepository.findAll();
+    public AuthorsDTO getAllAuthors(){
+        var authorsDTO = new AuthorsDTO();
+        authorRepository.findAll().stream()
+                .map(converter::entityToDto)
+                .forEach(authorsDTO::addDTO);
+        return authorsDTO;
     }
 
-    public List<Author> getAuthorsOfPaper(Long paperID){
+    public AuthorsDTO getAuthorsOfPaper(AuthorDTO authorDTO){
+        var paperID = converter.dtoToEntity(authorDTO).getPaper().getId();
         var paper = paperRepository.findById(paperID).orElseThrow(
                 () -> new InvalidIDException("Invalid Paper ID: " + paperID)
         );
-        return authorRepository.findAuthorsByPaper(paper);
+        var authorsDTO = new AuthorsDTO();
+        authorRepository.findAuthorsByPaper(paper).stream()
+            .map(converter::entityToDto)
+            .forEach(authorsDTO::addDTO);
+        return authorsDTO;
     }
 
-    public Author getAuthorByID(Long authorID){
-        return authorRepository.findById(authorID).orElseThrow(
+    public AuthorDTO getAuthorByID(Long authorID){
+        var author =  authorRepository.findById(authorID).orElseThrow(
                 () -> new InvalidIDException("Invalid Author ID: " + authorID)
         );
+        return converter.entityToDto(author);
     }
 
-    public Author getAuthorByEmail(String email){
-        return authorRepository.findAuthorByEmail(email).orElseThrow(
+    public AuthorDTO getAuthorByEmail(AuthorDTO authorDTO){
+        var email = authorDTO.getEmail();
```

Comment:
Here also we should do the conversion and not call the method from the dto, maybe we could change the controller and get only the email

Reply:
The issues with sending just the email woudl be that we also have a find by username function and we would end up with 2 conflicting endpotins (2 get endpoints with a string path variable). Would it be ok is we have 2 endpoints, like authors/email and authors/username?

Reply:
You could do it like that